BraveShared/Javascript/BrowserifiedJS.swift

```diff
@@ -0,0 +1,31 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import JavaScriptCore
+
+public typealias BundledBrowserifiedScript = BrowserifyExposable & BundledJSProtocol
+
+/// A wrapper around `JSContext` to help with browserified code.
+public class BrowserifiedJS {
+    public let context: JSContext
+    
+    public init?(browserifiedScript script: BundledBrowserifiedScript?) {
+        context = JSContext()
+        
+        guard let script = script, let scriptString = script.get else { return nil }
+        let src = "var window = this; \(scriptString)"
```

Comment:
Why do we need to set window here? Assuming it is due to the API that sync interacts with, but should add a why comment here.

Reply:
Browserify doesn't work otherwise, I will add a comment